src/main/java/com/gmail/goosius/siegewar/playeractions/PlaceBlock.java

```diff
@@ -57,6 +60,92 @@ public static void evaluateSiegeWarPlaceBlockRequest(Player player, Block block,
+					return; //SW doesn't un-cancel events in unsieged towns
+					
+				//Ensure player is on the town-hostile siege side				
+				Resident resident = TownyAPI.getInstance().getResident(player);
+				if(resident == null)
+					return;
+				if(!resident.hasNation())
+					return;
+				Siege siege = SiegeController.getSiege(town);
+				SiegeSide playerSiegeSide = SiegeWarAllegianceUtil.calculateCandidateSiegePlayerSide(player, resident.getTownOrNull(), siege);
+				if(playerSiegeSide == SiegeSide.NOBODY)
+					return; 
+				switch(siege.getSiegeType()) {
+					case CONQUEST:
+					case SUPPRESSION:
+						if(playerSiegeSide != SiegeSide.ATTACKERS)
+							return;
+						break;
+					case REVOLT:
+					case LIBERATION:
+						if(playerSiegeSide != SiegeSide.DEFENDERS)
+							return;
+						break;
+				}
+
+				//Ensure player has permission
+				if (!TownyUniverse.getInstance().getPermissionSource().testPermission(event.getPlayer(), SiegeWarPermissionNodes.SIEGEWAR_NATION_SIEGE_USE_BREACH_POINTS.getNode())) {
+					event.setMessage(Translation.of("msg_err_action_disable"));
+					return; 
+				}
+									
+				//Ensure there are enough breach points				
+				if(siege.getWallBreachPoints() < SiegeWarSettings.getWallBreachingBlockPlacementCost()) {			
+					event.setMessage(Translation.of("msg_err_not_enough_breach_points_for_action", SiegeWarSettings.getWallBreachingBlockPlacementCost(), siege.getFormattedBreachPoints()));
+					return;
+				}			
+
+				//Ensure the height is ok
+				if(SiegeWarDistanceUtil.isBlockCloseToTownBlock(block, town.getHomeBlockOrNull(), 2)) {					
+					int heightOfBlockRelativeToSiegeBanner = block.getY() - siege.getFlagLocation().getBlockY();
+					if(heightOfBlockRelativeToSiegeBanner < SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMin()) {					
+						event.setMessage(Translation.of("msg_err_cannot_place_at_this_height", SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMin(), SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMax()));
+						return;
+					}
+					if(heightOfBlockRelativeToSiegeBanner > SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMax()) {
+						event.setMessage(Translation.of("msg_err_cannot_place_at_this_height", SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMin(), SiegeWarSettings.getWallBreachingHomeblockBreachHeightLimitMax()));
+						return;
+					}	
+				}
+
+				//Ensure the material is ok to place
+				boolean allowedMaterial = false;
+				for(String materialString: SiegeWarSettings.getWallBreachingPlaceBlocksWhitelist()) {
+					if(materialString.startsWith("endswith=")) {
+						materialString = materialString.replace("endswith=", "");
+						if(event.getMaterial().name().toLowerCase().endsWith(materialString.toLowerCase())) {
+							allowedMaterial = true;
+							break;
+						}
+					}					
+					if(event.getMaterial().name().equalsIgnoreCase(materialString)) {
+						allowedMaterial = true;
+						break;
+					}
+				}
```

Comment:
Similar to the destroyblock class, this should be abstracted to a new method and return something using a cache or enumset, so that we aren't parsing over the list for every block break.

Reply:
Will do

Reply:
Done

Reply:
Done